src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -738,6 +773,9 @@ int dhcp_server_handle_message(sd_dhcp_server *server, DHCPMessage *message,
         existing_lease = hashmap_get(server->leases_by_client_id,
                                      &req->client_id);
 
+        static_lease = hashmap_get(server->static_leases_by_client_id,
```

Comment:
Not really a remark just a thought:
Have you thought about what it would take to add another match instead of just MACAddress? This looks like we'd just short-circut on a series of lookup functions and take the first result here? I am thinking about matching fields such as "Subscriber-ID" here.

Reply:
I agree, and I have reworked the parsing before with this kind of changes in mind. However, I would rather leave this change for another PR.